src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2232,24 +2237,112 @@ private void VisitBinaryOperatorChildren(BoundBinaryOperator node)
             stack.Free();
         }
 
+#nullable enable
         protected virtual void VisitBinaryOperatorChildren(ArrayBuilder<BoundBinaryOperator> stack)
         {
             var binary = stack.Pop();
-            VisitRvalue(binary.Left);
 
-            while (true)
+            // Only the leftmost operator of a left-associative binary operator chain can learn from a conditional access on the left
+            // For simplicity, we just special case it here.
+            // For example, `a?.b(out x) == true` has a conditional access on the left of the operator,
+            // but `expr == a?.b(out x) == true` has a conditional access on the right of the operator
+            if (VisitPossibleConditionalAccess(binary.Left, out var stateWhenNotNull)
+                && canLearnFromOperator(binary)
+                && isKnownNullOrNotNull(binary.Right))
             {
                 VisitRvalue(binary.Right);
+                Meet(ref stateWhenNotNull, ref State);
+                var isNullConstant = binary.Right.ConstantValue?.IsNull == true;
+                SetConditionalState(isNullConstant == isEquals(binary)
+                    ? (State, stateWhenNotNull)
+                    : (stateWhenNotNull, State));
+
+                if (stack.Count == 0)
+                {
+                    return;
+                }
+
+                binary = stack.Pop();
+            }
+
+            while (true)
+            {
+                if (!canLearnFromOperator(binary)
+                    || !learnFromOperator(binary))
+                {
+                    Unsplit();
+                    Visit(binary.Right);
+                }
 
                 if (stack.Count == 0)
                 {
                     break;
                 }
 
-                Unsplit(); // VisitRvalue does this
                 binary = stack.Pop();
             }
+
+            static bool canLearnFromOperator(BoundBinaryOperator binary)
+            {
+                var kind = binary.OperatorKind;
+                return kind.Operator() is BinaryOperatorKind.Equal or BinaryOperatorKind.NotEqual
+                    && (!kind.IsUserDefined() || kind.IsLifted());
+            }
+
+            static bool isKnownNullOrNotNull(BoundExpression expr)
+            {
+                return expr.ConstantValue is object
+                    || (expr is BoundConversion
+                        {
+                            ConversionKind: ConversionKind.ExplicitNullable or ConversionKind.ImplicitNullable,
+                        } conv && conv.Operand.Type!.IsNonNullableValueType());
```

Comment:
>!. [](start = 51, length = 2)

Why is this safe?

Reply:
The ExplicitNullable and ImplicitNullable conversions are only defined on operands with types, so if we get a conversion node with one of these kinds whose operand has no type, it means the conversion is malformed.

https://github.com/dotnet/csharplang/blob/main/spec/conversions.md#implicit-nullable-conversions
https://github.com/dotnet/csharplang/blob/main/spec/conversions.md#explicit-nullable-conversions

Reply:
Have expanded a bit on the test `EqualsCondAccess_12` to show that the `null` literal for example is not subject to this conversion (`null` being the most common typeless expression that we might think would be subject to this conversion).